### ISSUE TITLE
Color information lost when writing certain file formats

### ISSUE THREAD

**Issue by gab1one** [knime-ip/knip#303]
Writing multi-chanel images `.jpg` and `.png` files with the image writer node results in those images being single channel instead of multi channel (gray scale).
### Steps to reproduce:
- get the Image reader example workflow from the example server: 
  `tutoknime://EXAMPLES/099_Community/02_ImageProcessing/Tutorials/Node%20Tutorials/01_ImageReader/01_1_ImageReader`
- Write the read images (3 channel RGB) with the image writer node as `.jpg` files.
- read those images with the image reader:
- Input Image:
  ![image000001](https://cloud.githubusercontent.com/assets/1938154/13505192/b2e84122-e13d-11e5-826f-d1ff88d17794.jpg)
- result Image
  ![image000001 jpg](https://cloud.githubusercontent.com/assets/1938154/13505217/c76b3942-e13d-11e5-8276-23a8cd168a13.jpg)

**Comment by beschu** [comment]
I created a workflow that checks for this bug and placed it on mekur. It is named: color-info-lost-3channelRGB-jpg-issue303

I took the above mentioned image series and tested it:
Writing as  a jpg is lossy, hence we need measurement to see if images are the same regardless of lossy writing. 

My idea was: 
- Subtract the original and written image, apply convolution and then a global thresholder with manually set value (around 55) to hopefully get an image filled with zeros
- Compare to a generated all-0 image and if they are the same then everything is ok. If they are not the same it most likely isn't the .jpg compressions fault but the fault, that some color info is lost or a channel is missing.

With that approach i couldn't reproduce this bug and everything seems to be fine.

**Comment by tibuch** [comment]
Thanks!

**Comment by dietzc** [comment]
I don't understand. The problem seems to be that a dimension is either lost or added, not about lossy content at all. Can you confirm that the dimensionality of the written image and of the image read back to KNIME are the same?

**Comment by tibuch** [comment]
@dietzc @beschu and I looked at the workflow of @beschu. The written jpg files get read again and there are no issues with the dimensions. Looks like this bug got fixed somehow or @gab1one meant something different?

**Comment by tibuch** [comment]
@beschu and I just encountered the following problems:

- read png (X, Y, Channel) -> write png (X, Y, Time) (Issue #160)
- read png (X, Y, Channel) -> write jpg (X, Y) (maybe this issue?)

**Comment by dietzc** [comment]
What we read shouldn't be an issue at all, as we transform the image (at the moment) anyway to our internal data-format. The dimensionality of the image after having it read it is way more important. Can you therefore please reproduce the problem with the `Image Generator` (which creates exactly the same type of images you would get when reading in the `png`) and create two reproducing workflows + open issues accordingly, if not already opened? All this cross-referencing of issues gets a bit confusing.

**Comment by tibuch** [comment]
I agree that not reading the images is the problem. As far as I can tell it is a writer problem. We only use the readers to see the result of the writer.

**Comment by beschu** [comment]
I created the two reproducing workflows and opened the issues here:
#446 Write UnsignedByteType as jpg
#447 Write UnsignedByteType as png

**Comment by dietzc** [comment]
can we close this issue then?
